### ISSUE TITLE
[4.0] Ordering [a11y]

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#29936]
Any admin view that is by default sorted by ordering or any view that is subsequently change to sort by ordering will spit out errors because there is an input without a label. That is easily fixed but it still would be impossible to change the ordering without using a mouse

**Comment by hans2103** [comment]
@brianteeman could you give me a link to any admin view that is by default sorted by ordering? Saves me time searching.

**Comment by brianteeman** [comment]
Workflows

**Comment by hans2103** [comment]
Workflows found... had to enable it first:

## enable Workflow
- Go to Joomla 4 Administrator > Content > Articles
- Click _Options_ in the top-right corner.
- Click _Integration_
- Enable the _Workflow_ feature.
- Save & Close
- Go to Joomla 4 Administrator > Content > Workflow

I cannot reproduce the issue. I can change the order of column Ordering just by using a keyboard. 
<img width="868" alt="Schermafdruk 2020-07-05 14 16 41" src="https://user-images.githubusercontent.com/639822/86532440-434ccf80-beca-11ea-8821-d50764d5a43b.png">

Using AXE plugin I do notice an error. The columnhead on the left of the _ordering_ column is missing a label. But that is not what you are describing as an issue here. Or am I reading it wrong. 


# Form elements should have a visible label
Ensures that every form element is not solely labeled using the title or aria-describedby attributes

Element location
`input[name="checkall-toggle"]`
Element source
`<input type="checkbox" name="checkall-toggle" value="" title="Check All Items" onclick="Joomla.checkAll(this)">`

**Comment by brianteeman** [comment]
The issue is that you can not change the ordering not which column it is sorted by . Create a second workflow and try to change the order of the workflows.

**Comment by hans2103** [comment]
aha.. moving one row after / before another when sorted by _Ordering_.
I will first have to find out how I can change ordering by keyboard before I can test your issue. 
Thank you.

**Comment by hans2103** [comment]
@brianteeman can you explain me how I should change the ordering by keyboard? Can't find out how.

**Comment by brianteeman** [comment]
I don't know how. That's the point

**Comment by hans2103** [comment]
In other words, we have to find a solution to make drag&drop possible by keyboard too.

**Comment by hans2103** [comment]
@brianteeman I'm not able to find a proper solution.. yet. Perhaps someone else can give this issue a look too.

**Comment by brianteeman** [comment]
> In other words, we have to find a solution to make drag&drop possible by keyboard too.

I think there is a better/more practical option.

The ordering column has a hidden input which is the order number. Instead of hiding it we can make it visible when the table is sorted by column and then re-ordering is simply a case of editing the value in the input.

It just comes down to how we sabe the new order. Either it could be when you exit each input or we display a save order button.

**Comment by Fedik** [comment]
Joomla 2.5 had a good ordering feature without drag&drop.

**Comment by brianteeman** [comment]
the problem with the old 2.5 system was that you could only move up/down one row at a time

there are js options available that support keyboard drag and drop

**Comment by Fedik** [comment]
> you could only move up/down one row at a time

There was an input for position you want, that can be faster than dragging the row especially for large distances.
Plus it should be even nicer for mobile, where drag&drop is horrible.

> there are js options available that support keyboard drag and drop

we have something like that for subform,

**Comment by brianteeman** [comment]
> > you could only move up/down one row at a time
> 
> There was an input for position you want, that can be faster than dragging the row especially for large distances.

true - i forgot that. 

Its still there its just hidden

<img width="592" height="739" alt="Image" src="https://github.com/user-attachments/assets/d8a75914-d7cf-4031-b0bd-76e415cc674c" />
